File: src/test/java/com/plog/testUtil/WithCustomMockUser.java
package com.plog.testUtil;

import org.springframework.security.test.context.support.WithSecurityContext;

import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

/**
 * 추후 spring security 가 작성되면 테스트 중 유저 정보를 가져올 때 사용하는 어노테이션
 *
 * @author jack8
 * @since 2026-01-20
 */
@Retention(RetentionPolicy.RUNTIME)
@WithSecurityContext(factory = WithSecurityMockUserContextFactory.class)
public @interface WithCustomMockUser {

    long userId() default 1L;

    String email() default "test@test.com";

    String nickname() default "tester";
}

File: src/test/kotlin/com/plog/testUtil/WithCustomMockUser.kt
package com.plog.testUtil

import org.springframework.security.test.context.support.WithSecurityContext
import kotlin.annotation.Retention

/**
 * 추후 spring security 가 작성되면 테스트 중 유저 정보를 가져올 때 사용하는 어노테이션
 *
 * @author jack8
 * @since 2026-01-20
 */
@Retention(AnnotationRetention.RUNTIME)
@WithSecurityContext(factory = WithSecurityMockUserContextFactory::class)
annotation class WithCustomMockUser(
    val userId: Long = 1L,
    val email: String = "test@test.com",
    val nickname: String = "tester"
)
